Src/ILGPU/Backends/OpenCL/CLTypeGenerator.cs

```diff
@@ -117,24 +118,30 @@ public sealed class CLTypeGenerator : DisposeBase, ICLTypeGenerator
         /// </summary>
         /// <param name="basicValueType">The basic-value type to resolve.</param>
         /// <returns>The resolved OpenCL type name.</returns>
-        public static string GetBasicValueType(BasicValueType basicValueType) =>
-            BasicTypeMapping[(int)basicValueType];
+        public string GetBasicValueType(BasicValueType basicValueType) =>
+            basicValueType == BasicValueType.Float16 && !Capabilities.Float16
```

Comment:
Suggestion: Maybe we can try to refactor these pieces as well 🤓

Reply:
No change. Unfortunately, this is using `BasicValueType`, while the function below is `ArithmeticBasicValueType`.